sim/src/tlv.rs

```diff
@@ -535,15 +589,21 @@ impl ManifestGen for TlvGen {
                 }
             }
 
+            let flag = TlvFlags::ENCRYPTED_AES256 as u32;
+            let aes256 = (self.get_flags() & flag) == flag;
+
             let derived_key = match agreement::agree_ephemeral(
                 pk, &peer_pubk, ring::error::Unspecified, |shared| {
                     let salt = hkdf::Salt::new(hkdf::HKDF_SHA256, &[]);
                     let prk = salt.extract(&shared);
-                    let okm = match prk.expand(&[b"MCUBoot_ECIES_v1"], OkmLen(48)) {
+                    let okm = if aes256 { match prk.expand(&[b"MCUBoot_ECIES_v1"], OkmLen(64)) {
                         Ok(okm) => okm,
                         Err(_) => panic!("Failed building HKDF OKM"),
-                    };
-                    let mut buf = [0u8; 48];
+                    } } else { match prk.expand(&[b"MCUBoot_ECIES_v1"], OkmLen(48)) {
+                        Ok(okm) => okm,
+                        Err(_) => panic!("Failed building HKDF OKM"),
+                    } };
```

Comment:
I think if you added something like `let okm_len = if aes256 { 64 } else { 48 };` in the line above `let okm = ...`, you could just use that value and this block would not need to be duplicated.